### PR TITLE
Remove Hash from Recently Edited Types in `HotShotEvent`

### PR DESCRIPTION
**Pull request by jparr721**
Closes #<ISSUE_NUMBER>

### This PR: 
We needed to have `Hash` for `HotShotEvent` to be able to use it in `harness.rs`, but now we do not need that, and we can remove the hash trait from the `View` type and downstream types.

Also updates the `all` predicate to not use the `HashSet`.

### This PR does not: 
This does not evaluate all types who include `Hash` as a result of HotShotEvent requiring it and, instead, it only removes it from the types that I know for sure don't need to implement it.

### CODE REVIEW

File: crates/testing/src/predicates/event.rs [hunk 1]
```diff
@@ -58,18 +58,22 @@ where
...
-                PredicateResult::Pass
-            } else {
-                PredicateResult::Incomplete
+
+    // Remove duplicate events.
+    let mut events = events;
+    events.dedup();
+
+    let function =
+        move |e: &Arc<HotShotEvent<TYPES>>| match events.iter().find(|x| *x == e.as_ref()) {
```
Comment (ss-es) [inline crates/testing/src/predicates/event.rs:67]: If i understand `find` correctly, I don't think this is right: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find

It looks like it clears the iterator until it finds the first match.

File: crates/testing/src/predicates/event.rs [hunk 2]
```diff
@@ -58,18 +58,22 @@ where
...
-    let function = move |e: &Arc<HotShotEvent<TYPES>>| match set.take(e.as_ref()) {
-        Some(_) => {
-            if set.is_empty() {
-                PredicateResult::Pass
-            } else {
-                PredicateResult::Incomplete
+
+    // Remove duplicate events.
+    let mut events = events;
+    events.dedup();
```
Comment (ss-es) [inline crates/testing/src/predicates/event.rs:64]: I don't think we want to de-duplicate events (this leads to confusing behavior when calling `all(..)`).

**Review by ss-es — CHANGES_REQUESTED** [review]

**Review by ss-es — COMMENTED** [review]
Actually, thinking about it: could we make `all(events)` just a wrapper for `all_predicates(events.map(exact))`?

File: crates/testing/src/predicates/event.rs (hunk 2 above)
Comment (jparr721) [reply crates/testing/src/predicates/event.rs:64]: Sounds good. The HashSet had automatically de-duped so I was trying to find the correct one.

File: crates/testing/src/predicates/event.rs (hunk 1 above)
Comment (jparr721) [reply crates/testing/src/predicates/event.rs]: Fixed by just mapping over the events as suggested below.

**Review by ss-es — APPROVED** [review]
Looks good to me!

**Pull request merged**